### ISSUE TITLE
[docs] Add more examples

### ISSUE THREAD

**Issue by martinszeltins** [victorgarciaesgi/regle#37]
### Reproduction

⠀

### Steps to reproduce the bug

Add more useful examples to the docs.

````
- DONE -
[x] Custom Rules
[x] Conditional Rules
[x] Collections
[x] Server Validation
[x] Required Indicators

- TO DO -
[ ] Nested Forms
    Deeply nested forms, arrays, how to access parent items in each rule etc.
[ ] Multi-component Forms
    This will be a very popular example, to showcase how to use Pinia store
    and multiple components for a form. Also could demonstrate how to pass props to components.
[ ] Custom Metadata
    How to add custom metadata info to fields
[ ] Advanced Example
````

### Expected behavior

⠀

### Actual behavior

⠀

**Comment by victorgarciaesgi** [comment]
Yep all of your list is planned for the advanced example  👍

**Comment by martinszeltins** [comment]
> Yep all of your list is planned for the advanced example 👍

I think it'd be a good idea to also have separate examples listed as users may not know that they are included inside the `advanced-example`.

**Comment by victorgarciaesgi** [comment]
Yep agree! The advanced is more to show a more real world example on how to structure your logic around Pinia, Components, props etc..!

**Comment by martinszeltins** [comment]
I think this isn't completed yet, we still have these examples to add later.

````
- TO DO -
[ ] Nested Forms
    Deeply nested forms, arrays, how to access parent items in each rule etc.
[ ] Multi-component Forms
    This will be a very popular example, to showcase how to use Pinia store
    and multiple components for a form. Also could demonstrate how to pass props to components.
[ ] Custom Metadata
    How to add custom metadata info to fields
[ ] Advanced Example
````

**Comment by genu** [comment]
Are Multi-component Forms currently possible, or schema composition?

**Comment by victorgarciaesgi** [comment]
Yes it's already possible! https://reglejs.dev/typescript/typing-props

**Comment by genu** [comment]
> Yes it's already possible! https://reglejs.dev/typescript/typing-props

Hmm, thanks for pointing me there.

I'm still not exactly sure how to work with the state when it is spread across components, for example. For example:

I'm testing regle in an OS project here: https://github.com/genu/team-generator/pull/268/files#diff-12ebdf3416745f58aa6c74198855590abd67f26937ed199ab7c8295f0077c47c

Is this the proper way to work with a sub type for `players` and `EditPlayerForm`?

**Comment by martinszeltins** [comment]
> Hi! If this is still an issue, I'll gladly work on some examples whenever I have time, just let me know what I could help with!

I think examples are always welcome, it really helps users to get started quickly and understand how to use it.

**Comment by victorgarciaesgi** [comment]
Yeah advanced array usage is super cool 👌

**Comment by victorgarciaesgi** [comment]
On a side note, current exemples may be polished, i'll clean them a little 👍
